esp/esdllib/esdl_def.cpp

```diff
@@ -1470,20 +1473,18 @@ IEsdlDefObjectIterator* EsdlDefinition::getDependencies( const char* service, St
         {
             methodDef = serviceDef->queryMethodByName(methods.item(i));
 
-            if( NULL == methodDef )
-            {
+            if(!methodDef)
                 throw( MakeStringException(0, "ESDL Method Definition not found for %s in service %s", methods.item(i), service) );
-            }
-
+            if ((flags & DEPFLAG_ECL_ONLY) && methodDef->getPropInt("ecl_hide"))
+                continue;
             methodArray.append( *methodDef );
         }
     }
 
     this->gatherMethodDependencies( *dependencies, methodArray, requestedVer, opts, flags );
 
-    // Put the highest level objects on last- the services
-    // Tony did say that these will be useful
-    if( serviceDef != NULL )
+    bool allTypes = !(flags & DEPFLAG_INCLUDE_TYPES); //not asking for any explicit types indicates all types
```

Comment:
flags is a parameter to the method, not the class.. so allTypes has to be calculated locally.